src/timer.rs

```diff
@@ -173,6 +197,33 @@ macro_rules! hal {
                 pub fn release(self) -> $TIMX {
                     self.stop().release()
                 }
+
+                /// Returns the number of microseconds since the last update event.
+                /// *NOTE:* This method is not a very good candidate to keep track of time, because
+                /// it is very easy to lose an update event.
+                pub fn micros_since(&self) -> u32 {
+                    let timer_clock = self.clk.0;
+                    let psc = u32(self.tim.psc.read().psc().bits());
+                    
+                    // freq_divider is always bigger than 0, since (psc + 1) is always less than
+                    // timer_clock
+                    let freq_divider = u64(timer_clock / (psc + 1));
+                    let cnt = u64(self.tim.cnt.read().cnt().bits());
+                    
+                    // It is safe to make this cast, because the maximum timer period in this HAL is
+                    // 1s (1Hz), then 1 second < (2^32 - 1) microseconds
+                    u32(1_000_000 * cnt / freq_divider).unwrap()
+                }
+
+                /// Resets the counter and generates an update event
```

Comment:
Is this comment out of date, if I'm reading the function correctly, it does not generate an update event